### DISCUSSION TITLE
Why does my code interfere with other extensions when accessing the page content data?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#944] [Ask a question] [answered]
I'm currently developing a Mastodonbot extension to automatically post new pages to a Mastodon instance. In my extension settings I would like to have the possibility to customize the post template with placeholders such as in the blog or edit extension. 

```
$this->yellow->system->setDefault("mastodonbotTemplate", "@title @url"); // @title for page title, @content for page excerpt, @url for page url.
```

Being mainly a evil copy/paste developer, I used the edit extension as an example to appropriately convert the placeholders into actual content. 

```
    // Handle content file changes
    public function onEditContentFile($page, $action, $email) {
        if ($action=="create" && !preg_match("/exclude/i", $page->get("mastodon"))) {
            $title = $page->get("title");
            $url = $this->yellow->lookup->normaliseUrl(
                $this->yellow->system->get("coreServerScheme"),
                $this->yellow->system->get("coreServerAddress"),
                $this->yellow->system->get("coreServerBase"),
                $page->location
            );
            $content = $this->yellow->toolbox->createTextDescription($page->getContentHtml(), $this->yellow->system->get("mastodonbotMaxChars"), true, "<!--more-->");
            $language = $page->get("language");
            $apiKey = $this->yellow->system->get("mastodonbotApiKey");
            $instance = "https://".$this->yellow->system->get("mastodonbotInstance");
            $visibility = $this->yellow->system->get("mastodonbotVisibility");
            $statusTemplate = $this->yellow->system->get("mastodonbotTemplate");
            $statusTemplate = preg_replace("/@title/i", $title, $statusTemplate);
            $statusTemplate = preg_replace("/@content/i", $content, $statusTemplate);
            $statusTemplate = preg_replace("/@url/i", $url, $statusTemplate);
            
            $statusData = array(
                "status" => $statusTemplate,
                "visibility" => $visibility,
                "language" => $language,
            );
            $mastodon = new MastodonAPI($apiKey, $instance);
            $mastodon->postStatus($statusData);
        }
    }    
```

[Full code can be found in this testing branch](https://github.com/schulle4u/yellow-mastodonbot/blob/testing/mastodonbot.php). 

The code partly works, but it somehow breaks the creation of new pages. Blog pages are created, but they are missing the appropriate date filename prefix and include wrong page settings, thus not appearing in my blogpages layout. Apparently my placeholders are interfering with another extension, and I'm a little lost in finding the right debug method as there are no errors that can be tracked down. Can someone help me? 

The log file contains old install information, but here it is anyway. 

```
2024-01-11 15:20:03 info Install Datenstrom Yellow 0.8.23, PHP 8.3.1, Built-in 8.3.1, Linux
2024-01-11 15:20:03 info Install extension 'Update 0.8.97'
2024-01-11 15:20:03 info Install extension 'Core 0.8.125'
2024-01-11 15:20:03 info Install extension 'Edit 0.8.77'
2024-01-11 15:20:03 info Install extension 'Generate 0.8.52'
2024-01-11 15:20:03 info Install extension 'Image 0.8.19'
2024-01-11 15:20:03 info Install extension 'Markdown 0.8.26'
2024-01-11 15:20:03 info Install extension 'Serve 0.8.24'
2024-01-11 15:20:03 info Install extension 'Stockholm 0.8.14'
2024-01-11 15:20:03 info Install extension 'German 0.8.43'
2024-01-11 15:20:03 info Install extension 'English 0.8.43'
2024-01-11 15:20:03 info Install extension 'Swedish 0.8.43'
2024-01-11 15:20:22 info Install extension 'Blog 0.8.30'
2024-01-11 15:20:22 info Add user 'Schulle'
2024-01-11 15:20:52 info Install extension 'Mastodonbot 0.8.22'
```

**Comment by schulle4u** [comment]
Seems I found the problem and updated the topic accordingly. 

`$content = $this->yellow->toolbox->createTextDescription($page->getContentHtml(), $this->yellow->system->get("mastodonbotMaxChars"), true, "<!--more-->");`

If I comment out this line, the extension no longer interferes when creating blog pages. But... why? As far as I understand the API, this line should create a text description from the newly created page and optionally send it to Mastodon if configured.

**Reply by pftnhr** [reply]
Hello Steffen,

blog posts are created when you use `$this->yellow->page->getContentHtml()`. However, no content is posted to Mastodon, just a strange URL like `https://example.com/example-com-2`.

**Comment by markseuffert** [comment, marked as answer]
The edit extension has been updated, I think we found a timing problem and moved the edit events to a later time.

**Comment by schulle4u** [comment]
Thank you, works as expected now. 👍
